pkg/spec/util.go

```diff
@@ -38,3 +46,28 @@ func isPVCDefined(volumes []VolumeClaim, name string) bool {
 	}
 	return false
 }
+
+func GetScheme() (*runtime.Scheme, error) {
+	// Initializing the scheme with the core v1 api
+	scheme := api.Scheme
+	// Adding the batch scheme to support Jobs
+
+	// TODO: find a way where we don't have to add batch/v1 to the v1 scheme,
+	// instead we should be able to have different scheme for different controllers
+	if err := batch_v1.AddToScheme(scheme); err != nil {
```

Comment:
not sure what is happening here

Reply:
added a comment before the function

Reply:
// GetScheme() returns runtime.Scheme with supported Kubernetes API resource
// definitions which Kedge supports right now.
// The core v1 scheme is first initialized and then other controllers' scheme
// is added to that scheme, e.g. batch/v1 scheme is added to add support for
// Jobs controller to the v1 Scheme.
// Also, (from upstream) Scheme defines methods for serializing and deserializing API objects, a type
// registry for converting group, version, and kind information to and from Go
// schemas, and mappings between Go schemas of different versions. A scheme is the
// foundation for a versioned API and versioned configuration over time.